virtcontainers/kata_agent.go

```diff
@@ -720,16 +720,16 @@ func (k *kataAgent) replaceOCIMountsForStorages(spec *specs.Spec, volumeStorages
 	return nil
 }
 
-func constraintGRPCSpec(grpcSpec *grpc.Spec, systemdCgroup bool) {
+func constraintGRPCSpec(grpcSpec *grpc.Spec, systemdCgroup bool, DisableGuestSeccomp bool) {
 	// Disable Hooks since they have been handled on the host and there is
 	// no reason to send them to the agent. It would make no sense to try
 	// to apply them on the guest.
 	grpcSpec.Hooks = nil
 
-	// Disable Seccomp since they cannot be handled properly by the agent
-	// until we provide a guest image with libseccomp support. More details
-	// here: https://github.com/kata-containers/agent/issues/104
-	grpcSpec.Linux.Seccomp = nil
+	// Set grpcSpec.Linux.Seccomp as per the runtime configuration file
+	if DisableGuestSeccomp == true {
```

Comment:
Since we have exported guest seccomp capability in kata-containers/agent#384, please make use of it when making decisions about seccomp on the runtime side.